### PR TITLE
feat(router): add 东方财富研究报告

### PR DESCRIPTION
**Pull request by syzq**
## 该 PR 相关 Issue / Involved Issue

Close #12309

## 路由地址示例 / Example for the Proposed Route(s)

```routes
/eastmoney/report/strategyreport
/eastmoney/report/macresearch
/eastmoney/report/brokerreport
```

## 新 RSS 路由检查列表 / New RSS Route Checklist
  
- [ ] 新的路由 New Route
  - [x] 跟随 [v2 路由规范](https://docs.rsshub.app/joinus/script-standard.html) Follows [v2 Script Standard](https://docs.rsshub.app/en/joinus/script-standard.html)
- [ ] 文档说明 Documentation
  - [x] 中文文档 CN
  - [ ] 英文文档 EN
- [ ] 全文获取 fulltext
  - [ ] 使用缓存 Use Cache
- [ ] 反爬/频率限制 anti-bot or rate limit?
  - [ ] 如果有, 是否有对应的措施? If yes, do your code reflect this sign?
- [ ] [日期和时间](https://docs.rsshub.app/joinus/pub-date.html) [date and time](https://docs.rsshub.app/en/joinus/pub-date.html)
  - [x] 可以解析 Parsed
  - [ ] 时区调整 Correct TimeZone
- [ ] 添加了新的包 New package added
- [ ] `Puppeteer`

## 说明 / Note

### CODE REVIEW

File: lib/v2/eastmoney/radar.js [hunk 1]
```diff
@@ -17,5 +17,13 @@ module.exports = {
                 target: (_param, url) => `/eastmoney/search/${new URL(url).searchParams.get('KeyWord')}`,
             },
         ],
+        data: [
```
Comment (TonyRL) [inline lib/v2/eastmoney/radar.js:20]: Please insert new entries in alphabetical order.

File: lib/v2/eastmoney/maintainer.js [hunk 2]
```diff
@@ -1,4 +1,5 @@
 module.exports = {
     '/ttjj/user/:uid': ['zidekuls'],
     '/search/:keyword': ['drgnchan'],
+    '/report/:category': ['syzq'],
```
Comment (TonyRL) [inline lib/v2/eastmoney/maintainer.js:4]: Please insert new entries in alphabetical order.

File: lib/v2/eastmoney/router.js [hunk 3]
```diff
@@ -1,4 +1,5 @@
 module.exports = (router) => {
     router.get('/ttjj/user/:uid', require('./ttjj/user'));
     router.get('/search/:keyword', require('./search/index'));
+    router.get('/report/:category', require('./report/index'));
```
Comment (TonyRL) [inline lib/v2/eastmoney/router.js:4]: Please insert new entries in alphabetical order.

File: lib/v2/eastmoney/report/index.js [hunk 4]
```diff
@@ -0,0 +1,53 @@
+const cheerio = require('cheerio');
```
Comment (TonyRL) [inline lib/v2/eastmoney/report/index.js:1]: No need to use puppeteer at all. The data is embedded as a JSON `initdata` in HTML.

**Review by TonyRL — COMMENTED** [review]
Please take a look at https://docs.rsshub.app/joinus/new-rss/submit-route.html and correct the PR title.
This is the first line of the PR template and should be visited if you have any trouble filling out the template and should not be deleted at all.

File: lib/v2/eastmoney/report/index.js (hunk 4 above)
Comment (syzq) [reply lib/v2/eastmoney/report/index.js:2]: ok

File: lib/v2/eastmoney/report/index.js [hunk 5]
```diff
@@ -0,0 +1,48 @@
...
+            if ($(item).text().includes('initdata')) {
+                const reg = /\[\{(.+)\}\]/g;
+                const dataTemp = $(item).text().match(reg);
+                listTemp = JSON.parse(dataTemp[0]);
+            }
+            return listTemp;
+        });
+    const filterList = initData.filter((item) => item.length !== 0)[0];
+
+    const list = filterList.map((item) => ({
```
Comment (TonyRL) [inline lib/v2/eastmoney/report/index.js:36]:
```suggestion
    const initData = JSON.parse(
        $('script')
            .text()
            .match(/var initdata = (.*?);/)[1]
    );

    const list = initData.data.map((item) => ({
```

File: lib/v2/eastmoney/report/index.js [hunk 6]
```diff
@@ -0,0 +1,48 @@
...
+        title: `【${item.orgSName}】- ${item.title}`,
+        link: `${baseUrl}/report/${linkType[category]}?encodeUrl=${item.encodeUrl}`,
+        pubDate: parseDate(item.publishDate),
+        author: item.author,
```
Comment (TonyRL) [inline lib/v2/eastmoney/report/index.js:40]: `author` needs to be a string instead of an array.

**Pull request merged**